packages/data-model/src/readable/data-instance.ts

```diff
@@ -0,0 +1,74 @@
+import { InteropFactory, ReadableDataRegistration } from '..';
+import { ReadableResource } from './resource';
+
+interface ChildInfo {
+  shapeTree: {
+    iri: string;
+    label: string;
+  };
+  count: number;
+}
+
+export class ReadableDataInstance extends ReadableResource {
+  dataRegistration: ReadableDataRegistration;
+
+  children: ChildInfo[];
+
+  constructor(public iri: string, public factory: InteropFactory, public descriptionLang?: string) {
+    super(iri, factory);
+  }
+
+  private async bootstrap(): Promise<void> {
+    await this.fetchData();
+    await this.buildDataRegistration();
+    if (this.descriptionLang) {
+      await this.dataRegistration?.shapeTree?.getDescription(this.descriptionLang);
```

Comment:
What is done with the result form the Promise here?

Reply:
the side effect will make the description in requested language available on the shape tree